src/Mod/Draft/DraftGeomUtils.py

```diff
@@ -1225,48 +1246,170 @@ def offsetWire(wire,dvec,bind=False,occ=False,widthList=None):
+            curOrientation = firstOrientation
+            curDir = firstDir
+            curAlign = firstAlign
+            delta = dvec
+
+        # record current edge's Orientation, and set Delta
+        if i != 0:  #else:
+            if isinstance(curredge.Curve,Part.Circle):				# TODO Should also calculate 1st edge direction above
+                delta = curredge.Vertexes[0].Point.sub(curredge.Curve.Center)
+            else:
+                delta = vec(curredge).cross(norm)
+            curOrientation = curredge.Vertexes[0].Orientation			# TODO Could be edge.Orientation in fact
+
+        # Consider individual edge width
+
+        if widthList:  # ArchWall should now always provide widthList
             try:
                 if widthList[i] > 0:
-                    delta = DraftVecUtils.scaleTo(dvec, widthList[i])
-                elif widthList[0] > 0:
-                    delta = DraftVecUtils.scaleTo(dvec, widthList[0])	 	# to follow widthList[0]
-
-                # i.e. if widthList[0] == 0, though widthList is not False
-                # but if dev is provided still, fallback to dvec
+                    delta = DraftVecUtils.scaleTo(delta, widthList[i])
                 elif dvec:
-                    delta = dvec
-
-                else:  
-                    return None
+                    delta = DraftVecUtils.scaleTo(delta, dvec.Length)
+                else:
+                    #just hardcoded default value as ArchWall would provide if dvec is not provided either
+                    delta = DraftVecUtils.scaleTo(delta, 200)
             except:
-                if widthList[0] > 0:
-                    delta = DraftVecUtils.scaleTo(dvec, widthList[0])	 	# to follow widthList[0]
-                delta = dvec
+                if dvec:
+                    delta = DraftVecUtils.scaleTo(delta, dvec.Length)
+                else:
+                    #just hardcoded default value as ArchWall would provide if dvec is not provided either
+                    delta = DraftVecUtils.scaleTo(delta, 200)
         else:
-              delta = dvec
+            delta = DraftVecUtils.scaleTo(delta,dvec.Length)
 
-        if i != 0:
-            if isinstance(curredge.Curve,Part.Circle):
-                v = curredge.tangentAt(curredge.FirstParameter)
-            else:
-                v = vec(curredge)
+        # Consider individual edge Align direction - ArchWall should now always provide alignList
 
-            ## TODO - 2019.6.16 - 'calculate' 'offset' direction (in vector) edge by edge instead of rotating previous vector based on dvec in future
-
-            angle = DraftVecUtils.angle(firstVec,v,norm)			# use vec deduced depending on geometry instead of - angle = DraftVecUtils.angle(vec(edges[0]),v,norm)
-            delta = DraftVecUtils.rotate(delta,angle,norm)
+        if i == 0:
+                if alignListC[0] == 'Center':
```

Comment:
You should check the indentation in here. It seems to be 8 spaces, while it should be only 4 spaces.